src/main/java/it/pagopa/pn/delivery/middleware/notificationdao/EntityToDtoNotificationMapper.java

```diff
@@ -0,0 +1,178 @@
+
+        NotificationPaymentInfo.NotificationPaymentInfoBuilder builder =  NotificationPaymentInfo.builder()
+                .iuv( entity.getIuv() )
+                .notificationFeePolicy( entity.getNotificationFeePolicy() )
+                ;
+
+        NotificationAttachment f24Digital = buildAttachment( entity.getF24DigitalKey(),
+                             entity.getF24DigitalVersionId(), entity.getF24DigitalDigestSha256() );
+        NotificationAttachment f24Analog = buildAttachment( entity.getF24AnalogKey(),
+                             entity.getF24AnalogVersionId(), entity.getF24AnalogDigestSha256() );
+        NotificationAttachment f24FlatRate = buildAttachment( entity.getF24FlatRateKey(),
+                           entity.getF24FlatRateVersionId(), entity.getF24FlatRateDigestSha256() );
+
+        boolean anyF24moduleNotNull = Stream.of( f24Analog, f24Digital, f24FlatRate )
+                .anyMatch( Objects::nonNull );
+
+        if( anyF24moduleNotNull ) {
+            builder.f24( NotificationPaymentInfo.F24.builder()
+                    .analog( f24Analog )
+                    .digital( f24Digital )
+                    .flatRate( f24FlatRate )
+                    .build()
+            );
+        }
+
+        return builder.build();
+    }
+
+    private NotificationAttachment buildAttachment(String key, String version, String sha256 ) {
+        NotificationAttachment result;
+        if ( StringUtils.isAllBlank( key, version, sha256 ) ) {
+            result = null;
+        }
+        else if ( version != null && StringUtils.isNotBlank( sha256 ) && StringUtils.isNotBlank( key ) ) {
+            result = NotificationAttachment.builder()
+                    .ref( NotificationAttachment.Ref.builder()
+                            .key( key )
+                            .versionToken( version )
+                            .build()
+                    )
+                    .digests( NotificationAttachment.Digests.builder()
+                            .sha256( sha256 )
+                            .build()
+                    )
+                    .build();
+        }
+        else {
+            throw new PnInternalException( "Error key (" + key + ") version (" + version + ") and sha256 (" + sha256 + ") are both required or both blank" );
+        }
+        return result;
+    }
+
+    private NotificationAttachment buildAttachment(String key, String version, String sha256, String contentType, String title ) {
+         NotificationAttachment notificationAttachment = buildAttachment(key, version, sha256);
+         if ( notificationAttachment != null ){
+             notificationAttachment = notificationAttachment.toBuilder()
+                     .contentType(contentType)
+                     .title( title )
+                     .build();
+         }
+        return notificationAttachment;
+    }
+
+    private List<NotificationAttachment> buildDocumentsList( NotificationEntity entity ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 136-143, lines 147-159 into separate methods.